### PR TITLE
Feature/host groups

### PR DESCRIPTION
**Pull request by vinay-ebi**
To help web manipulate the DB copy service to release the datasets available on staging, the concept of "group" of server could be handy, now we can assign host servers to particular group

### CODE REVIEW

File: ensembl_dbcopy/admin.py [hunk 1]
```diff
@@ -19,11 +19,29 @@
...
+
+@receiver(post_delete, sender=UsersGroup)
+def delete_group(sender, instance, **kwargs):
+    user_groups = [ obj.name for obj in UsersGroup.objects.all() ]
+    host_groups = [ obj.group_name for obj in Group.objects.all() ]
+    new_usergroup = list( set(host_groups) - set(user_groups) )
+    for each_group in new_usergroup:
+        Group.objects.filter(group_name=each_group).delete()
 
 class GroupInlineForm(forms.ModelForm):
```
Comment (marcoooo) [inline ensembl_dbcopy/admin.py:46]: in a perfect world, this class would reside in a `forms.py` file.

File: ensembl_dbcopy/admin.py [hunk 2]
```diff
@@ -51,10 +69,15 @@ class Media:
...
+    fields = ('name', 'port', 'mysql_user', 'virtual_machine', 'mysqld_file_owner','groups', 'active')
+    search_fields = ('name', 'port', 'mysql_user', 'virtual_machine', 'mysqld_file_owner', 'active')
+    
+    def get_groups(self, obj):
+        return ", ".join([str(g) for g in obj.groups.all()])
+
+    get_groups.short_description = 'Groups '
 
 
 class OverallStatusFilter(SimpleListFilter):
```
Comment (marcoooo) [inline ensembl_dbcopy/admin.py:83]: And this one in a file names `filters.py` :-)

File: ensembl_dbcopy/models.py [hunk 3]
```diff
@@ -171,5 +174,7 @@ class Meta:
         verbose_name = 'Host Group'
 
     group_id = models.BigAutoField(primary_key=True)
-    host_id = models.ForeignKey(Host, db_column='auto_id', on_delete=models.CASCADE, related_name='groups')
+    #host_id = models.ForeignKey(Host, db_column='auto_id', on_delete=models.CASCADE, related_name='groups')
```
Comment (marcoooo) [inline ensembl_dbcopy/models.py:177]: You shouldn't be using the same group table. going to be a mess in the long run. This initial group model is supposedly dedicated to "Users's group" and determine the user's group permissions on a host.

File: ensembl_dbcopy/admin.py (hunk 2 above)
Comment (vinay-ebi) [reply ensembl_dbcopy/admin.py:83]: sure i will do that

File: ensembl_dbcopy/models.py (hunk 3 above)
Comment (vinay-ebi) [reply ensembl_dbcopy/models.py:177]: should i creatge new table named 'hostTogroup'  and this should be in sync with auth_group table in production_service db

**Comment by vinay-ebi** [comment]
closing the draft PR.
New PR rainsed : https://github.com/Ensembl/ensembl-production-services/pull/119

**Pull request closed without merging**